lib/ecto/association.ex

```diff
@@ -233,6 +233,18 @@ defmodule Ecto.Association do
               {[changeset|changesets], structs, halt, valid?}
             {:ok, struct} ->
               {[changeset|changesets], [struct | structs], halt, valid?}
+            {:error, {error_changeset, :insert_join}} ->
+              # Roll the errors originating from a join into the parent_changeset
+              case changesets do
```

Comment:
I'm not sure at which point in the code I can update the parent changeset (the one which contains the constraint and from which the insert_join originates).